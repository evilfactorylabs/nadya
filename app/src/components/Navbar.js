import React, { useState, useEffect } from 'react'
import dayjs from 'dayjs'

import { navigate } from '@reach/router'
import { makeStyles } from '@material-ui/core/styles'
import { deepOrange } from '@material-ui/core/colors'

import SwipeableDrawer from '@material-ui/core/SwipeableDrawer'
import Avatar from '@material-ui/core/Avatar'
import AppBar from '@material-ui/core/AppBar'
import Toolbar from '@material-ui/core/Toolbar'
import Typography from '@material-ui/core/Typography'
import CardHeader from '@material-ui/core/CardHeader'
import IconButton from '@material-ui/core/IconButton'
import BackIcon from '@material-ui/icons/ArrowBack'
import MenuIcon from '@material-ui/icons/Menu'

import List from '@material-ui/core/List'
import ListItem from '@material-ui/core/ListItem'
import ListItemText from '@material-ui/core/ListItemText'
import LogoType from '../assets/logotype.png'

import { APP_VER } from '../constants'
import { getUser, getUserIdFromLS } from 'services/user'

const useStyles = makeStyles(theme => ({
  appVer: {
    position: 'absolute',
    bottom: 10,
    left: 10
  },
  list: {
    width: 250
  },
  root: {
    flexGrow: 1
  },
  avatar: {
    color: '#fff',
    backgroundColor: deepOrange[500]
  },
  title: {
    flexGrow: 1
  },
  logoContainer: {
    margin: 'auto',
    paddingRight: theme.spacing(4)
  },
  logoType: {
    float: 'left',
    width: 160
  }
}))

export default ({ shouldUseBackIcon }) => {
  const classes = useStyles()
  const userId = getUserIdFromLS()
  const [drawerOpen, setDrawerOpen] = useState(false)
  const [user, setUser] = useState({})

  useEffect(() => {
    getUser(userId).then(user => {
      setUser(user)
    })
  }, [userId])

  const handleClick = () => {
    if (shouldUseBackIcon) {
      if (window.history.state) {
        window.history.back()
      } else {
        navigate('/')
      }
    } else {
      setDrawerOpen(true)
    }
  }

  return (
    <div className={classes.root}>
      <AppBar position='static'>
        <SwipeableDrawer
          open={drawerOpen}
          onClose={() => setDrawerOpen(false)}
          onOpen={() => setDrawerOpen(true)}
        >
          <div className={classes.list} onClick={() => setDrawerOpen(false)}>
            <CardHeader
              avatar={
                <Avatar className={classes.avatar}>
                  {user && user.name ? user.name.charAt(0).toUpperCase() : 'N'}
                </Avatar>
              }
              title={user.name || 'Nadya User'}
              subheader={`Daftar pada ${dayjs(user.created_at).format(
                'DD MMM YYYY'
              )}`}
            />
            <List>
              <ListItem button onClick={() => navigate('/')}>
                <ListItemText primary='Subscription saya' />
              </ListItem>
              <ListItem button onClick={() => navigate('/onboarding')}>
                <ListItemText primary='Tentang' />
              </ListItem>
              <ListItem button onClick={() => navigate('/setting')}>
                <ListItemText>Pengaturan</ListItemText>
              </ListItem>
            </List>
            <Typography
              variant='body2'
              color='textSecondary'
              className={classes.appVer}
            >
              App ver: {APP_VER}
            </Typography>
          </div>
        </SwipeableDrawer>
        <Toolbar>
          <IconButton
            onClick={handleClick}
            edge='start'
            className={classes.menuButton}
            color='inherit'
            aria-label='Menu'
          >
            {shouldUseBackIcon ? <BackIcon /> : <MenuIcon />}
          </IconButton>
          <div className={classes.logoContainer}>
            <img alt='nadya logo' className={classes.logoType} src={LogoType} />
          </div>
        </Toolbar>
      </AppBar>
    </div>
  )
}
