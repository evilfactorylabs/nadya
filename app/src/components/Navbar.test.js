import Navbar from './Navbar'

afterAll(async () => {
  await cleanup()
})

describe('Fragments: Navbar.js', () => {
  test('It should render Navbar screen', () => {
    const { container } = render(<Navbar />)

    expect(container).toBeInTheDocument()
  })
})
