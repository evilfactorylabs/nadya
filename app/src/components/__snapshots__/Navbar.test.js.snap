// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fragments: Navbar.js It should render Navbar screen 1`] = `
<div>
  <div
    class="makeStyles-root-3"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary"
    >
      <div
        class="PrivateSwipeArea-root-81 PrivateSwipeArea-anchorLeft-82"
        style="width: 20px;"
      />
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <button
          aria-label="Menu"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <div
          class="makeStyles-logoContainer-6"
        >
          <img
            alt="nadya logo"
            class="makeStyles-logoType-7"
            src="logotype.png"
          />
        </div>
      </div>
    </header>
  </div>
</div>
`;
